landsat/anaconda-project.yml

```diff
@@ -10,38 +10,30 @@ examples_config:
 
 user_fields: [examples_config]
 
-channels: [pyviz, defaults]
+channels: [pyviz, defaults, conda-forge]
```

Comment:
Did you add `conda-forge` for `rioxarray`? If so, I'd probably suggest inlining it with `- conda-forge::rioxarary` instead of adding it globally. I've seen `channel::package` being recommended in some places, I'm not entirely sure how it works, e.g. does it pull all the dependencies of `rioxarray` from conda-forge or can it pull them from the global channels too?

Well but to be honest I think that instead of that I would just make `conda-forge` the unique and global channel:
- there's no reason to depend on `pyviz` I think, all the HoloViz packages are on conda-forge, just some hours after they're published on `pyviz`
- `defaults` can be quite lagging, e.g. rasterio 1.3.7 is on conda-forge vs. `1.2.10` on defaults (which is not too bad, but still behind). That doesn't matter so much for these examples (v.s. when we're running a test suite) but still, why not using the latest when we can.
- `conda-forge` works pretty well and with the spread of the libmamba solver users don't have too wait so much when solving complex environments

On that last point, I'm actually thinking about something else where @jlstevens could shed some light on. IIRC `anaconda-project` does solve again based on the list of dependencies pinned in the lock file, instead of installing them all directly. Could it be that when `conda-forge` is involved, installing a locked project can take a very long time when you're not using the libmamba solver? In which case @sandhujasmine forget about what I said on `conda-forge`, as I think we expect some of the projects on examples to be re-used by AE5 users who can't switch easily to libmamba.

Reply:
Note that anaconda-project is completely misleading about the :: syntax. It does *not*, as you would hope and expect, bring in just that package, or even bring in just that package and its dependencies. Last I checked, it simply added the channel listed onto the channel search list, which to me does not seem like reasonable behavior.  That's another reason to simply use c-f if you do need anything from c-f.

Reply:
> Note that anaconda-project is completely misleading about the :: syntax. It does not, as you would hope and

Indeed - I'm not sure I understand channel priority or how it is used to solve for the packages; probably not easy to state. I had to limit it to c-f. I'm not sure how to interpret / understand :: syntax. I found few references to this syntax - saw this on [conda-managing-channels](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-channels.html):

"Essentially, the order goes: channelA::numpy-1.13_1 > channelA::numpy-1.12.1_1 > channelA::numpy-1.12.1_0 > channelB::numpy-1.13_1" 

Will ask folks who know more about conda.

Reply:
Note that this behavior is controlled by anaconda-project, not conda. conda does behave properly with channel specifications like that, but anaconda-project (last I looked) simply strips off the channel specifier, adds it to the list of allowed channels, and only then gets conda involved. So don't look at what conda does for ::, because that's not invoked here.